spec/fx/adapters/postgres_spec.rb

```diff
@@ -20,6 +20,38 @@ module Fx::Adapters
       end
     end
 
+    describe "#update_function" do
+      let(:name) { "test" }
+      let(:sql_definition) do
+        <<-EOS
+          CREATE OR REPLACE FUNCTION test()
+          RETURNS text AS $$
+          BEGIN
+              RETURN 'test';
+          END;
+          $$ LANGUAGE plpgsql;
+        EOS
+      end
+
+      it "successfully updates a function" do
+        adapter = Postgres.new
+
+        expect(adapter).to receive(:drop_function).with(name)
```

Comment:
Can you split this up into an `allow` step and then `expect`? Look elsewhere in the application for how I normally like to do that.

Reply:
This is also about duplication:

```ruby
        adapter = Postgres.new

        expect(adapter).not_to receive(:drop_function)
        expect(adapter).to receive(:create_function).with(sql_definition)

        adapter.update_function(name, sql_definition, replace: true)
```

VS:

```ruby
        allow(adapter).not_to receive(:drop_function)
        allow(adapter).to receive(:create_function).with(sql_definition)

        adapter = Postgres.new
        adapter.update_function(name, sql_definition, replace: true)

        expect(adapter).not_to have_received(:drop_function)
        expect(adapter).to have_received(:create_function).with(sql_definition)
```

Reply:
I want it duplicated to keep each test more readable

Reply:
Try to follow the style of the tests already in place